katas/design-patterns/chatroom.js

```diff
@@ -0,0 +1,58 @@
+var Participant = function (name) {
+    this.name = name;
+    this.chatroom = null;
+};
+
+Participant.prototype = {
+    send: function (message, to) {
+        this.chatroom.send(message, this, to);
+    },
+    receive: function (message, from) {
+        console.log(from.name + ' to ' + this.name + ': ' + message);
+    }
+};
+
+var Chatroom = function () {
+    var participants = {};
+
+    return {
+        register: function (participant) {
+            participants[participant.name] = participant;
+            participant.chatroom = this;
+        },
+
+        send: function (message, from, to) {
+            var key;
+            if (to) {                      // single message
+                to.receive(message, from);
+            } else {                       // broadcast message
+                for (key in participants) {
+                    if (participants[key] !== from) {
+                        participants[key].receive(message, from);
+                    }
+                }
+            }
+        }
+    };
+};
+
+function run() {
```

Comment:
add a space before ()

Reply:
moved the file into the folder, no issue detected by eslint with adding a space before a parenthesis, but the space was added